sfputil/main.py

```diff
@@ -917,28 +917,25 @@ def fetch_error_status_from_platform_api(port):
 
     output_list = output.split('\n')
     for output_str in output_list:
-        # The output of all SFP error status are a list consisting of element with convention of '<sfp no>:<error status>'
+        # The output of all SFP error status are a list consisting of <error status> sorted physical port indices
         # Besides, there can be some logs captured during the platform API executing
         # So, first of all, we need to skip all the logs until find the output list of SFP error status
         if output_str[0] == '[' and output_str[-1] == ']':
             output_list = ast.literal_eval(output_str)
             break
 
-    output_dict = {}
-    for output in output_list:
-        sfp_index, error_status = output.split(':')
-        output_dict[int(sfp_index)] = error_status
+    physical_port_set = set()
+    for port in logical_port_list: 
+        physical_port_set.add(logical_port_to_physical_port_index(port))
 
-    output = []
-    for logical_port_name in logical_port_list:
-        physical_port_list = logical_port_name_to_physical_port_list(logical_port_name)
-        port_name = get_physical_port_name(logical_port_name, 1, False)
-
-        if is_port_type_rj45(logical_port_name):
-            output.append([port_name, "N/A"])
-        else:
-            output.append([port_name, output_dict.get(physical_port_list[0])])
+    physical_port_list = list(physical_port_set)
+    # Map of <physical port# : error status> 
+    error_map = {physical_port_list[i] : output_list[i] for i in range(len(physical_port_list))}
```

Comment:
This is still problematic when the list physical_port_list is not continuous.
This is because the index of physical_port_list is used as the index of the output_list. 
In the below example, physical_port 8, and 10 don't exist but output_list still contains error_description of all ports, including 8 and 10. As a result, the error descriptions get misaligned from physical_port 9:
- the error description of physical_port 9 should be output_list[8] which is `OK` but it is output_list[7] which is `Unplugged` where 7 is the index of `9` in physical_port_list
- similar as physical_port 11.
```
> /usr/local/lib/python3.9/dist-packages/sfputil/main.py(933)<dictcomp>()
-> error_map = {physical_port_list[i] : output_list[i] for i in range(len(physical_port_list))}
(Pdb) p error_map
{1: 'OK', 2: 'OK', 3: 'OK', 4: 'OK', 5: 'OK', 6: 'OK', 7: 'OK', 9: 'Unplugged', 11: 'OK', 12: 'Unplugged', 13: 'OK', 14: 'OK', 15: 'OK', 16: 'OK', 17: 'OK', 18: 'OK', 19: 'OK', 20: 'OK', 21: 'OK', 22: 'OK', 23: 'OK', 24: 'OK', 25: 'OK', 26: 'OK', 27: 'OK', 28: 'OK', 29: 'OK', 30: 'OK', 31: 'OK', 32: 'OK'}
(Pdb) p physical_port_list
[1, 2, 3, 4, 5, 6, 7, 9, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32]
(Pdb) p output_list
['OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'Unplugged', 'OK', 'Unplugged', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK', 'OK']
```